### PR TITLE
Reduce RBAC permissions for the various components

### LINKED ISSUE #3041: Remove unnecessary RBAC permission in helm charts
**What would you like to be added**:

Remove all the unnecessary RBAC permission in the helm charts.

#1105 may relate to this.

**Why is this needed**:

Unnecessary rbac permissions can lead to security risks. Currently, the submariner-operator's helm charts have applied for too many permissions it doesn’t need, such as Deployment `submariner-operator`. Among them, we found that several sensitive permissions may even lead to the hijacking of the cluster under specific attacks. Due to the risk of security disclosure, we have hidden the details of these permissions in the public issue. We have reported this security issue through private email and received confirmation from the community.

### PR DESCRIPTION
**Pull request by tpantelis**
A recent post to the **submariner-security** outlined potential vulnerabilities regarding submariner's RBAC permissions being too broad:

- create/patch/update verb of the statefulsets/replicasets resource
   - A malicious user can create a privileged container with a malicious container image capable of container escape. This would allow him/she gaining ROOT privilege of the worker node the created container deployed. Since the malicious user can control the pod scheduling parameters (e.g., replicas number, node affinity, …), he/she should be able to deploy the malicious container to every (or almost every) worker node. This means the malicious user will be able to control every (or almost every) worker node in the cluster.
   - The other permissions of statefulsets/replicasets/deployments/daemonsets also share the same risks.

In addition, many permissions aren't even used/needed. Also specifying a wildcard for resources and verbs is not best practice - Sonar [flags this](https://sonarcloud.io/project/issues?resolved=false&types=VULNERABILITY&id=submariner-io_submariner-operator&open=AY7GnmQtf1gbRJmNPGKK).

The intent of this PR is to reduce the permissions to only what's needed. See individual commits for details.

Note that for these changes to actually be applied requires updating and rebuilding **subctl**. However, I mirrored the changes in https://github.com/submariner-io/submariner-charts/pull/508 and the E2E in the **submariner-charts** repo directly tests the changes in the PR. So we can be pretty confident the changes are accurate.

Fixes https://github.com/submariner-io/submariner-operator/issues/3041

Depends on https://github.com/submariner-io/admiral/pull/905

### CODE REVIEW

**Review by aswinsuryan — APPROVED** [review]

**Review by skitt — CHANGES_REQUESTED** [review]
This needs some autosquashing ;-).

**Review by kaaass — APPROVED** [review]

**Comment by tpantelis** [comment]
I added https://github.com/submariner-io/submariner-operator/pull/3040/commits/2691550c7a23ea59f93e98275d0b609cf878834c to eliminate the operator `namespaces` permission.

Also added https://github.com/submariner-io/submariner-operator/pull/3040/commits/b268cd5e097e32387512926cc9a18b208531e2d6 and https://github.com/submariner-io/submariner-operator/pull/3040/commits/4b7e3e23d88770a6d5fa76300d31a24f97567bd6.

File: go.mod [hunk 1]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.73.2
 	github.com/prometheus-operator/prometheus-operator/pkg/client v0.73.2
 	github.com/prometheus/client_golang v1.19.0
-	github.com/submariner-io/admiral v0.18.0-m3
+	github.com/submariner-io/admiral v0.18.0-m3.0.20240504143439-22ee51817b45
```
Comment (skitt) [inline go.mod]: This will need to be updated once https://github.com/submariner-io/admiral/pull/905 is merged.

**Review by skitt — CHANGES_REQUESTED** [review]
Requesting changes just to make sure the Admiral import is updated to match the merged commit after https://github.com/submariner-io/admiral/pull/905 goes in.

**Comment by tpantelis** [comment]
> Requesting changes just to make sure the Admiral import is updated to match the merged commit after [submariner-io/admiral#905](https://github.com/submariner-io/admiral/pull/905) goes in.

Done

**Review by skitt — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Pull request merged**